From 91ae11776f0f8acbafa0f388dd4305483170fa64 Mon Sep 17 00:00:00 2001
From: Bala Sai Kosuri <bkosuri@qti.qualcomm.com>
Date: Thu, 17 Sep 2026 16:01:03 +0530
Subject: [PATCH 1/2] pulse: NULL check before de-reference

The new pulsedirectsink element calls gst_pulse_fill_format_info
with a NULL channels pointer (it only needs the pa_format_info,
not the channel count). Guard the write to *channels to avoid a
NULL dereference.

Upstream-Status: Submitted [https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/12144]

Signed-off-by: Bala Sai Kosuri <bkosuri@qti.qualcomm.com>
---
 ext/pulse/pulseutil.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ext/pulse/pulseutil.c b/ext/pulse/pulseutil.c
index 588b882..64f9533 100644
--- a/ext/pulse/pulseutil.c
+++ b/ext/pulse/pulseutil.c
@@ -188,7 +188,8 @@ gst_pulse_fill_format_info (GstAudioRingBufferSpec * spec, pa_format_info ** f,
     goto fail;
 
   *f = format;
-  *channels = GST_AUDIO_INFO_CHANNELS (ainfo);
+  if (channels)
+    *channels = GST_AUDIO_INFO_CHANNELS (ainfo);
 
   return TRUE;
 
-- 
2.34.1

